Source/DiffableCollection/FetchedCollectionDiffableDataSource.swift

```diff
@@ -0,0 +1,206 @@
+		} else {
+			externalSnapshot.reloadItems(itemsToReconfigure)
+		}
+
+		if isUpdatingAutomatically {
+			applySnapshot(snapshot: externalSnapshot)
+		} else {
+			pendingSnapshot = externalSnapshot
+		}
+	}
+
+	// MARK: - Helpers
+
+	/// Propagates changes from fetched results controllers into the internal snapshot.
+	private func updateInternalSnapshot(withChangedContent snapshot: NSDiffableDataSourceSnapshot<String, FetchedDiffableItem>, maintainSectionOrder: Bool) {
+		let indices: [String: Int]
+		if maintainSectionOrder { // maintain section order according to `internalSnapshot`
+			indices = snapshot.sectionIdentifiers.reduce(into: [:]) { result, sectionIdentifier in
+				if let index = internalSnapshot.sectionIdentifiers.firstIndex(where: { $0.identifier == sectionIdentifier }) {
+					result[sectionIdentifier] = index
+				}
+			}
+		} else { // maintain section order according to `snapshot`
+			indices = [:]
+		}
+
+		// delete all sections, as they will be inserted again anyway
+		// this will prevent the issue where the last item from a section is not being removed properly
+		internalSnapshot.deleteSections(internalSnapshot.sectionIdentifiers)
+
+		// add all sections and items
+		snapshot.sectionIdentifiers.forEach { sectionIdentifier in
+			let section: FetchedDiffableSection = .init(identifier: sectionIdentifier)
+			let items: [FetchedDiffableItem] = snapshot.itemIdentifiers(inSection: sectionIdentifier)
+			if (isHidingEmptySections && !items.isEmpty) || !isHidingEmptySections {
+				if let index = indices[sectionIdentifier], index < internalSnapshot.numberOfSections {
+					let sectionAtIndex = internalSnapshot.sectionIdentifiers[index]
+					internalSnapshot.insertSections([section], beforeSection: sectionAtIndex)
+				} else {
+					internalSnapshot.appendSections([section])
+				}
+				internalSnapshot.appendItems(items, toSection: section)
+			}
+		}
+	}
+
+	/// Transforms the `NSDiffableDataSourceSnapshotReference` into a `NSDiffableDataSourceSnapshot` object.
+	private func transform(snapshot: NSDiffableDataSourceSnapshotReference, context: NSManagedObjectContext) -> NSDiffableDataSourceSnapshot<String, FetchedDiffableItem> {
+		obtainPermanentIDs(for: snapshot.transform(), context: context)
+	}
+
+	/// The napshot returned by the `NSFetchedResultsController` instance contains temporary `NSManagedObjectID`s.
+	/// Working with temporary identifiers can lead to issues since at some point in time they will no longer exist.
+	private func obtainPermanentIDs(for snapshot: NSDiffableDataSourceSnapshot<String, FetchedDiffableItem>, context: NSManagedObjectContext) -> NSDiffableDataSourceSnapshot<String, FetchedDiffableItem> {
+		guard snapshot.itemIdentifiers.contains(where: { $0.object is NSManagedObjectID }) else { return snapshot }
+		var snapshotWithPermanentIDs: NSDiffableDataSourceSnapshot<String, FetchedDiffableItem> = .init()
+		snapshot.sectionIdentifiers.forEach { sectionIdentifier in
```

Comment:
I think some properties are ignored when copying items from one snapshot to another.

- reconfiguredItems
- reloadItems
- reloadSectionIdentifiers